pool/pool.go

```diff
@@ -541,19 +550,52 @@ func (p *pool) checkSessionTokenErr(err error, address string) bool {
 	return false
 }
 
+func (p *pool) removeSessionTokenAfterThreshold(cfg *callConfig) error {
+	cp, err := p.connection()
+	if err != nil {
+		return err
+	}
+
+	key := p.key
+	if cfg.key != nil {
+		key = cfg.key
+	}
+
+	ts, ok := p.cache.GetAccessTime(formCacheKey(cp.address, key))
+	if ok && time.Since(ts) > p.stokenTh {
+		p.cache.DeleteByPrefix(cp.address)
+	}
+
+	return nil
+}
+
 func (p *pool) PutObject(ctx context.Context, params *client.PutObjectParams, opts ...CallOption) (*oid.ID, error) {
 	cfg := cfgFromOpts(append(opts, useDefaultSession())...)
+
+	// Put object is  different  from  other object  service methods, because
```

Comment:
// Put object is different from other object service methods. Put
	// request can't be resent in case of session token failures (i.e. session
	// token is invalid due to lifetime expiration or server restart). The reason
	// is that object's payload can be provided as a stream that should be
	// read only once.
	//
	// To solve this issue, pool regenerates session tokens upon each request.
	// In case of subsequent requests, pool avoids session token initialization
	// by checking when the session token was accessed for the last time.
	// If it hits a threshold, session token is removed from cache for a new one to be issued.